exchanges/asset/asset.go

```diff
@@ -91,26 +110,72 @@ func (a Items) JoinToString(separator string) string {
 // IsValid returns whether or not the supplied asset type is valid or
 // not
 func (a Item) IsValid() bool {
-	for x := range supported {
-		if supported[x].String() == a.String() {
-			return true
-		}
+	return a != Empty && supportedFlag&a == a
+}
+
+// UnmarshalJSON comforms type to the umarshaler interface
+func (a *Item) UnmarshalJSON(d []byte) error {
+	var assetString string
+	err := json.Unmarshal(d, &assetString)
+	if err != nil {
+		return err
 	}
-	return false
+
+	if assetString == "" {
+		return nil
+	}
+
+	ai, err := New(assetString)
+	if err != nil {
+		return err
+	}
+
+	*a = ai
+	return nil
+}
+
+// MarshalJSON comforms type to the marshaller interface
+func (a Item) MarshalJSON() ([]byte, error) {
+	if a == Empty {
+		return json.Marshal("")
+	}
+	return json.Marshal(a.String())
 }
 
 // New takes an input matches to relevant package assets
 func New(input string) (Item, error) {
 	input = strings.ToLower(input)
-	for i := range supported {
-		if string(supported[i]) == input {
-			return supported[i], nil
-		}
+	switch input {
+	case "spot":
+		return Spot, nil
+	case "margin":
+		return Margin, nil
+	case "marginfunding":
+		return MarginFunding, nil
+	case "index":
+		return Index, nil
+	case "binary":
+		return Binary, nil
+	case "perpetualcontract":
+		return PerpetualContract, nil
+	case "perpetualswap":
+		return PerpetualSwap, nil
+	case "futures":
+		return Futures, nil
+	case "upsideprofitcontract":
+		return UpsideProfitContract, nil
+	case "downsideprofitcontract":
+		return DownsideProfitContract, nil
+	case "coinmarginedfutures":
+		return CoinMarginedFutures, nil
+	case "usdtmarginedfutures":
+		return USDTMarginedFutures, nil
+	default:
+		return 0, fmt.Errorf("%w %v, only supports %s",
```

Comment:
```suggestion
		return 0, fmt.Errorf("%w '%v', only supports %s",
```
Helps highlight empty string instead of looking weird